sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/SnapshotSelector.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.Data.AppConfiguration
+{
+    /// <summary>
+    /// <see cref="SnapshotSelector"/> is a set of options that allows selecting a filtered set of <see cref="ConfigurationSnapshot"/> entities.
+    /// </summary>
+    public class SnapshotSelector
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="SnapshotSelector"/>.
+        /// </summary>
+        /// <param name="name">A filter for the name of the returned snapshots.</param>
+        /// <param name="fields">A list of fields used to specify which fields are included in the returned resource(s).</param>
+        /// <param name="status">A list of snapshot status used to filter the returned snapshots based on their status property.</param>
+        public SnapshotSelector(string name = null, IEnumerable<SnapshotFields> fields = null, IEnumerable<ConfigurationSnapshotStatus> status = null)
+        {
+            Name = name;
+            Fields = fields?.ToList();
+            Status = status?.ToList();
```

Comment:
```suggestion
            Fields = fields?.ToList() ?? new List<SnapshotFields>();
            Status = status?.ToList() ?? new List<ConfigurationSnapshotStatus>();
```

This seems like an unusual pattern.   Don't we normally prefer creating model types with an empty default list and allowing values to be set using initializer syntax?

I think we'll want to check this with the architects before merging.

Reply:
Yes, I agree. I incorporated this approach because the `GetSnapshots` method permits passing null values for the name, list of fields, and list of status parameters. When I initially implemented your recommended pattern, my fields and status were never null. After reviewing our standard approach, I noticed that we typically use `ChangeTrackingList` instead of `List`. Therefore, by defining `new ChangeTrackingList<SnapshotFields>();`, our inner list remains empty, and we send null for list types.

I have addressed the [SnapshotSelector](https://github.com/Azure/azure-sdk-for-net/blob/08c3f5c53279f5a4383b296ca200946e4b5f7063/sdk/appconfiguration/Azure.Data.AppConfiguration/src/Models/SnapshotSelector.cs#L18-L22) type using our standard approach for non-required fields.